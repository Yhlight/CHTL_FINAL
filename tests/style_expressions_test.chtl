// Test file for CHTL Style Expressions

#box1 {
    style {
        width: 100px;
        height: 50px;
        padding: 10px;
    }
}

#box2 {
    style {
        // Simple Arithmetic
        width: 10px + 20px; // 30px
        height: 50px - 10px; // 40px
        left: 10px * 3; // 30px
        top: 100px / 2; // 50px

        // Operator Precedence
        margin: 10px + 5px * 2; // 20px

        // Parentheses
        padding: (10px + 5px) * 2; // 30px

        // Property Reference
        font-size: #box1.padding; // 10px

        // Property Reference in Expression
        z-index: #box1.width + #box1.height; // 150

        // Conditional Expression
        color: 10 > 5 ? "red" : "blue"; // red
        background-color: #box1.width == 100px ? "green" : "yellow"; // green
    }
}
