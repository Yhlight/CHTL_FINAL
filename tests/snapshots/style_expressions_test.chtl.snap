Successfully compiled tests/style_expressions_test.chtl to output.html
